ethereum/contracts/IncentivizedInboundChannel.sol

```diff
@@ -54,66 +55,37 @@ contract IncentivizedInboundChannel is AccessControl {
         renounceRole(DEFAULT_ADMIN_ROLE, msg.sender);
     }
 
-    // TODO: Submit should take in all inputs required for verification,
-    // including eg: _parachainBlockNumber, _parachainMerkleProof, parachainHeadsMMRProof
     function submit(
         Message[] calldata _messages,
-        bytes32 _commitment,
-        bytes32 _beefyMMRLeaf,
+        ParachainLightClient.OwnParachainHeadPartial
+            calldata _ownParachainHeadPartial,
+        bytes32[] calldata _parachainHeadsProof,
+        ParachainLightClient.BeefyMMRLeafPartial calldata _beefyMMRLeafPartial,
         uint256 _beefyMMRLeafIndex,
         uint256 _beefyMMRLeafCount,
-        bytes32[] memory _beefyMMRLeafProof
+        bytes32[] calldata _beefyMMRLeafProof
```

Comment:
Shall we wrap all of these types in a simple `ParachainLightClient.CommitmentProof` struct? That would simplify the API here a bit.

Reply:
yeh probably a good idea to do in a follow up PR